repronim/retrace/rpzutil.py

```diff
@@ -0,0 +1,166 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil; coding: utf-8 -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the repronim package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Functions to read and manipulate reprozip yaml files
+
+"""
+from __future__ import unicode_literals
+import datetime
+import repronim
+import repronim.utils as utils
+import yaml
+import io
+import repronim.retrace.packagemanagers as packagemanagers
+
+
+def read_reprozip_yaml(filename):
+    """Parses a ReproZip YAML file into a configuration object
+
+    Given the path to a ReproZip YAML file, it parses the file and
+    stores the resulting environment configuration into an object.
+
+    Parameters
+    ----------
+    filename : str
+        Path to the ReproZip YAML file
+
+    Return
+    ------
+    dict
+        The environment configuration
+
+    """
+    with io.open(filename, encoding='utf-8') as fp:
+        config = yaml.safe_load(fp)
+        # TODO: Check version of ReproZip file and warn if unknown
+        return config
+
+
+def identify_packages(config):
+    """Identifies packages in the current environment from a ReproZip config
+
+    Given a ReproZip configuration, it analyzes the current environment to
+    find details about the source packages, and places the results back into
+    the configuration object.
+
+    Parameters
+    ----------
+    config : dict
+        ReproZip configuration (input/output)
+
+    Return
+    ------
+    dict
+        A reference to the input dict
+
+    """
+    # Immediately clone the configuration
+    files = get_system_files(config)
+
+    (packages, unidentified_files) = packagemanagers.identify_packages(
+        list(files))
+
+    # Update reprozip package assignment
+    config['packages'] = packages
+
+    # set any files not identified
+    config['other_files'] = list(unidentified_files)
+    config['other_files'].sort()
+
+    return config
+
+
+def write_config(os, config):
+    """Writes an environment config to a stream
+
+    Parameters
+    ----------
+    os
+        Output Stream
+
+    config : dict
+        Environment configuration (input)
+
+    """
+    envconfig = dict(config)  # Shallow copy for destruction
+    os.write(("# ReproNim Environment Configuration File\n" +
```

Comment:
`os` is one of the builtin modules -- I got really confused at first ;)  'stream' or `output` could be a better name